### PR TITLE
RTC and I2C-RTC modules

### PR DESCRIPTION
**Pull request by DanielHernandez-0801**

### CODE REVIEW

File: firmware/stm32f4_disc/Core/Src/i2c-rtc.c
Comment (Matt-Lemcke) [inline firmware/stm32f4_disc/Core/Src/i2c-rtc.c]: Remove the date, month, dow and year inputs

File: firmware/inc/rtc.h
Comment (Matt-Lemcke) [inline firmware/inc/rtc.h]: Type name shouldn't be all caps, just the enum value ids. Rename to AlarmStatus

File: firmware/inc/rtc.h
Comment (Matt-Lemcke) [inline firmware/inc/rtc.h]: Same as above

File: firmware/inc/rtc.h
Comment (Matt-Lemcke) [inline firmware/inc/rtc.h]: We can't use types only defined in this module as parameters for function pointers from other modules. These will be populated with functions from i2c-rtc.h and the enableAlarm function takes a uint8_t and bool as inputs. Types have to match these functions

File: firmware/inc/rtc.h
Comment (Matt-Lemcke) [inline firmware/inc/rtc.h]: Same as above for lines 37 and 47-49

File: firmware/src/rtc.c
Comment (Matt-Lemcke) [inline firmware/src/rtc.c]: Technically don't need the id!=ALARM id!=TIMER check since only parameters of type ALARM_ID can be passed in. Up to you if you want to keep it

**Review by Matt-Lemcke — CHANGES_REQUESTED** [review]
Some minor fixes. Otherwise looks good if it's been tested on hardware

File: firmware/src/rtc.c
Comment (DanielHernandez-0801) [reply firmware/src/rtc.c]: You're right, the less code the better so I will take it out

**Comment by DanielHernandez-0801** [comment]
Above comments resolved. Hold off on merging until properly tested on hardware. Will update when testing is done.

**Review by Matt-Lemcke — APPROVED** [review]
Approved pending hardware testing

**Comment by DanielHernandez-0801** [comment]
Passed all hardware tests. Ready for merging.

**Pull request merged**
